src/trix/models/document.coffee

```diff
@@ -125,17 +125,34 @@ class Trix.Document extends Trix.Object
 
     text = leftText.appendText(rightText)
 
-    removingLeftBlock = leftIndex isnt rightIndex and leftLocation.offset is 0
-    useRightBlock = removingLeftBlock and leftBlock.getAttributeLevel() >= rightBlock.getAttributeLevel()
-
-    if useRightBlock
+    previousCharacterRange = [startPosition - 1, endPosition - 1]
+    previousCharacter = rightBlock.text.getTextAtRange(previousCharacterRange).toString()
+    currentCharacter = @getCharacterAtPosition(startPosition, endPosition)
+    nextCharacter = @getCharacterAtPosition(endPosition, endPosition + 1)
```

Comment:
`range` could be any size `>= 1` so declaring these won't always be a single character.